File: common/src/main/java/com/ssblur/scriptor/feature/ScriptorInscriptionsFeature.java
package com.ssblur.scriptor.feature;

import com.mojang.serialization.Codec;
import com.ssblur.scriptor.block.GenerateBlock;
import net.minecraft.world.level.levelgen.feature.Feature;
import net.minecraft.world.level.levelgen.feature.FeaturePlaceContext;
import net.minecraft.world.level.levelgen.feature.configurations.NoneFeatureConfiguration;

public class ScriptorInscriptionsFeature extends Feature<NoneFeatureConfiguration> {
  public ScriptorInscriptionsFeature(Codec<NoneFeatureConfiguration> codec) {
    super(codec);
  }

  @Override
  public boolean place(FeaturePlaceContext<NoneFeatureConfiguration> context) {
    var level = context.level();
    var pos = context.origin();

    level.setBlock(pos, GenerateBlock.generateEngraving(), 0);

    return true;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/feature/ScriptorInscriptionsFeature.kt
package com.ssblur.scriptor.feature

import com.mojang.serialization.Codec
import com.ssblur.scriptor.block.GenerateBlock
import net.minecraft.world.level.levelgen.feature.Feature
import net.minecraft.world.level.levelgen.feature.FeaturePlaceContext
import net.minecraft.world.level.levelgen.feature.configurations.NoneFeatureConfiguration

class ScriptorInscriptionsFeature(codec: Codec<NoneFeatureConfiguration?>?) :
    Feature<NoneFeatureConfiguration?>(codec) {
    override fun place(context: FeaturePlaceContext<NoneFeatureConfiguration?>): Boolean {
        val level = context.level()
        val pos = context.origin()

        level.setBlock(pos, GenerateBlock.generateEngraving(), 0)

        return true
    }
}
